### PR TITLE
uv python install: Detect hard-float support

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

## Summary

This pull request fixes [#6873](https://github.com/astral-sh/uv/issues/6873) by implementing
a mechanism to detect whether the ARM architecture supports the VFP (Vector Floating Point) hardware,
which is critical for selecting the appropriate `libc` variant.

The detection is based on reading the flags from `/proc/cpuinfo`.
Specifically, if the `vfp` flag is present, the system will use the `gnueabihf` (hard-float ABI) variant;
if not, it will default to `gnueabi` (soft-float ABI).

For additional context, see the [Debian documentation on the ARM Hard Float Port](https://wiki.debian.org/ArmHardFloatPort#VFP).

## Test Plan

- [ ] Verify that the correct `libc` variant is selected on an ARM system with and without VFP support.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -2676,6 +2726,32 @@ dependencies = [
...
 ]
 
+[[package]]
+name = "procfs"
+version = "0.16.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "731e0d9356b0c25f16f33b5be79b1c57b562f141ebfcdb0ad8ac2c13a24293b4"
+dependencies = [
+ "bitflags 2.6.0",
+ "chrono",
```
Comment (charliermarsh) [inline Cargo.lock:2736]: Can we disable the optional `chrono` feature?

File: crates/uv-python/src/cpuinfo.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+use anyhow::Error;
+#[cfg(target_os = "linux")]
+use procfs::{CpuInfo, Current};
+
+#[cfg(target_os = "linux")]
+pub(crate) fn detect_hardware_floating_point_support() -> Result<bool, Error> {
+    let cpu_info = CpuInfo::current()?;
+    for num in 0..cpu_info.num_cores() {
+        if let Some(flags) = cpu_info.flags(num) {
+            if flags.contains(&"vfp") {
```
Comment (konstin) [inline crates/uv-python/src/cpuinfo.rs:12]: Can you link the debian docs page here?

File: crates/uv-python/src/cpuinfo.rs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+//! Fetches CPU information.
+
+use anyhow::Error;
+#[cfg(target_os = "linux")]
+use procfs::{CpuInfo, Current};
+
+#[cfg(target_os = "linux")]
+pub(crate) fn detect_hardware_floating_point_support() -> Result<bool, Error> {
```
Comment (konstin) [inline crates/uv-python/src/cpuinfo.rs:8]: The docstring should mention something about arm, afaik this instruction set does not exist on x86

**Review by konstin — APPROVED** [review]
Looks good!

File: crates/uv-python/src/cpuinfo.rs (hunk 3 above)
Comment (kakkoyun) [reply crates/uv-python/src/cpuinfo.rs:8]: That's true. I will add the documentation.

**Comment by kakkoyun** [comment]
Quick update: I'm still trying to set up some old RPIs to test this.

**Comment by zanieb** [comment]
Replacing with https://github.com/astral-sh/uv/pull/8498

**Pull request closed without merging**